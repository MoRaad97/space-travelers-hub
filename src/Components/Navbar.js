import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../imgs/planet.png';
import './style/Navbar.css';

const Navbar = () => {
  const navLinkStyle = ({ isActive }) => ({ textDecoration: isActive ? 'underline' : 'none' });
  return (
    <nav className="navbar">
      <div className="container">
        <div className="logo">
          <img src={logo} alt="logo" />
          <h1>Space Travelers&apos; Hub</h1>
        </div>
        <ul>
          <NavLink style={navLinkStyle} to="/Rockets">Rockets</NavLink>
          <NavLink style={navLinkStyle} to="/Missions">Missions</NavLink>
          <NavLink style={navLinkStyle} to="/">My Profile</NavLink>
        </ul>
      </div>
    </nav>

  );
};

export default Navbar;
