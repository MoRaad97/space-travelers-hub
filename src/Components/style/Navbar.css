nav {
  font-family: Arial, Helvetica, sans-serif;
  border-bottom: solid 1px black;
}

nav h1 {
  font-size: 28px;
}

nav img {
  width: 75px;
  margin-right: 10px;
}

nav ul {
  margin: 0;
  padding: 0;
  list-style: none;
}

nav a {
  font-size: 17px;
  color: #2196f3;
  text-decoration: none;
}

nav a:active {
  color: rgb(0, 213, 255);
}

nav a:hover {
  color: blue;
}

nav ul a:nth-child(2) {
  padding: 0 22px 0 22px;
  position: relative;
}

nav ul a:nth-child(2)::after {
  content: "";
  width: 2px;
  height: 17px;
  background-color: black;
  display: inline-block;
  position: absolute;
  right: 10px;
  bottom: 2px;
}

nav .logo {
  display: flex;
  justify-content: space-between;
}

nav .container {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
