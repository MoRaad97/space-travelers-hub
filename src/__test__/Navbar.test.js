import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../Components/Navbar';

it('Navbar Component snapshot test', () => {
  const NavbarRender = renderer
    .create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    )
    .toJSON();
  expect(NavbarRender).toMatchSnapshot();
});
