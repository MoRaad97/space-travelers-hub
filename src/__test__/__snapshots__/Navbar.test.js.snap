// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component snapshot test 1`] = `
<nav
  className="navbar"
>
  <div
    className="container"
  >
    <div
      className="logo"
    >
      <img
        alt="logo"
        src="planet.png"
      />
      <h1>
        Space Travelers' Hub
      </h1>
    </div>
    <ul>
      <a
        className=""
        href="/Rockets"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Rockets
      </a>
      <a
        className=""
        href="/Missions"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Missions
      </a>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "underline",
          }
        }
      >
        My Profile
      </a>
    </ul>
  </div>
</nav>
`;
